applications/StructuralMechanicsApplication/custom_elements/cr_beam_element_linear_2D2N.cpp

```diff
@@ -0,0 +1,249 @@
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:     BSD License
+//  license: 	 structural_mechanics_application/license.txt
+//
+//  Main authors: Klaus B. Sautter
+//                   
+//                   
+//
+// System includes
+
+// External includes
+
+// Project includes
+#include "custom_elements/cr_beam_element_linear_2D2N.hpp"
+#include "custom_utilities/static_condensation_utility.h"
+#include "structural_mechanics_application_variables.h"
+#include "includes/define.h"
+
+
+namespace Kratos
+{
+
+	CrBeamElementLinear2D2N::CrBeamElementLinear2D2N(IndexType NewId,
+		GeometryType::Pointer pGeometry)
+		: CrBeamElement2D2N(NewId, pGeometry)
+	{}
+
+	CrBeamElementLinear2D2N::CrBeamElementLinear2D2N(IndexType NewId,
+		GeometryType::Pointer pGeometry,
+		PropertiesType::Pointer pProperties)
+		: CrBeamElement2D2N(NewId, pGeometry, pProperties)
+	{}
+
+	Element::Pointer CrBeamElementLinear2D2N::Create(IndexType NewId,
+		NodesArrayType const& rThisNodes,
+		PropertiesType::Pointer pProperties) const
+	{
+		const GeometryType& rGeom = this->GetGeometry();
+		return Kratos::make_shared<CrBeamElementLinear2D2N>(
+            NewId, rGeom.Create(rThisNodes), pProperties);
+	}
+
+	CrBeamElementLinear2D2N::~CrBeamElementLinear2D2N() {}
+
+	
+
+	void CrBeamElementLinear2D2N::CalculateLocalSystem( MatrixType& rLeftHandSideMatrix,
+		VectorType& rRightHandSideVector,ProcessInfo& rCurrentProcessInfo)
+		{
+			KRATOS_TRY;
+			// Kt
+			this->CalculateLeftHandSide(rLeftHandSideMatrix,rCurrentProcessInfo);
+				
+			Vector NodalDeformation = ZeroVector(msElementSize);
+			this->GetValuesVector(NodalDeformation);
+			rRightHandSideVector = ZeroVector(msElementSize);
+			rRightHandSideVector -= prod(rLeftHandSideMatrix, NodalDeformation);
+	
+			rRightHandSideVector += this->CalculateBodyForces();
+
+			KRATOS_CATCH("")
+		}
+
+	void CrBeamElementLinear2D2N::CalculateRightHandSide(VectorType& rRightHandSideVector,
+		ProcessInfo& rCurrentProcessInfo)
+		{
+			KRATOS_TRY;
+			Vector NodalDeformation = ZeroVector(msElementSize);
+			this->GetValuesVector(NodalDeformation);
+			rRightHandSideVector = ZeroVector(msElementSize);
+			rRightHandSideVector -= prod(this->K_master, NodalDeformation);
+
+
+
+			rRightHandSideVector += this->CalculateBodyForces();
+			KRATOS_CATCH("")
+		}
+
+	void CrBeamElementLinear2D2N::CalculateLeftHandSide(MatrixType& rLeftHandSideMatrix,
+		ProcessInfo& rCurrentProcessInfo) 
+		{
+			KRATOS_TRY;
+			rLeftHandSideMatrix = this->CreateElementStiffnessMatrix_Total();
+
+
+			//// start static condensation
+			if (this->Has(CONDENSED_DOF_LIST))
+			{
+				Vector dof_list_input = this->GetValue(CONDENSED_DOF_LIST);
+				std::vector<int> dofList(dof_list_input.size());
+				for (size_t i=0;i<dof_list_input.size();++i) dofList[i]=dof_list_input[i];
+				StaticCondensationUtility::CondenseLeftHandSide(*this,rLeftHandSideMatrix,dofList);

```

Comment:
indentiation

Reply:
where exactly?

Reply:
Now I don't see it any more
Maybe you can use the link I sent you by mail, where one can replace tabs with spaces in vscode